presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java

```diff
@@ -364,8 +369,15 @@ public void start()
                     return;
                 }
 
-                // analyze query
-                PlanRoot plan = analyzeQuery();
+                PlanRoot plan;
+
+                // setup a thread watchdog in case query analyzer ends up in an infinite loop
+                try (ThreadWatchdog w = new ThreadWatchdog(Thread.currentThread(),
```

Comment:
- move `Thread.currentThread()` to its own line. We leave the first line empty for function parameters if too long.
- instead of calling it `w`, we could call it `unused` or `thread`. Usually we spell the full name for variables and definitions